applications/sandbox/systems/physics_test_system.cpp

```diff
@@ -0,0 +1,729 @@
+
+            //get support point for the tangents and inverse tangents of this face 
+            PhysicsStatics::GetSupportPoint(faceVertices, faceForward, maxForward);
+            PhysicsStatics::GetSupportPoint(faceVertices, -faceRight, minForward);
+
+            PhysicsStatics::GetSupportPoint(faceVertices, faceRight, maxRight);
+            PhysicsStatics::GetSupportPoint(faceVertices, -faceRight, minRight);
+
+            float maxForwardLength = math::dot(maxForward - faceCentroid, faceForward);
+            float minForwardLength = math::dot(minForward - faceCentroid, -faceForward);
+
+            float maxRightLength = math::dot(maxRight - faceCentroid, faceRight);
+            float minRightLength = math::dot(minRight - faceCentroid, -faceRight);
+
+            math::vec3 min = faceCentroid - (faceForward * minForwardLength) - (faceRight * minRightLength);
+            float forwardLength = (minForwardLength + maxForwardLength);
+            float rightLength =  (minRightLength + maxRightLength);
+
+            auto calculateUV = [&min,&faceForward,&faceRight,&forwardLength,&rightLength](math::vec3 edgePosition)->math::vec2
+            {
+                math::vec2 result;
+                result.x = math::dot(edgePosition - min, faceForward)/ forwardLength;
+                result.y = math::dot(edgePosition - min, faceRight)/ rightLength;
+
+                return result;
+            };
+
+            auto populateMesh = [&calculateUV,&vertices,&normals,&uvs](HalfEdgeEdge* edge)
+            {
+                math::vec3 normal = edge->face->normal;
+               
+                vertices.push_back(edge->edgePosition);
+                uvs.push_back(calculateUV(edge->edgePosition));
+                normals.push_back(normal);
+
+                vertices.push_back(edge->nextEdge->edgePosition);
+                uvs.push_back(calculateUV(edge->nextEdge->edgePosition));
+                normals.push_back(normal);
+
+                vertices.push_back(edge->face->centroid);
+                uvs.push_back(calculateUV(edge->face->centroid));
+                normals.push_back(normal);
+            };
+
+            face->forEachEdge(populateMesh);
+        }
+
+        for (int i = 0; i < vertices.size(); i++)
+        {
+            indices.push_back(i);
+        }
+
+        for (int i = 0; i < vertices.size(); i++)
+        {
+            colors.push_back(math::color(1.0f, 1.0f, 1.0f));
+        }
+
+        newMesh.calculate_tangents(&newMesh);
+
+        sub_mesh newSubMesh;
+        newSubMesh.indexCount = newMesh.indices.size();
+        newSubMesh.indexOffset = 0;
+        newSubMesh.materialIndex = 0;
+
+        newMesh.submeshes.push_back(newSubMesh);
+
+        static int procCount = 0;
+        std::string name = std::string("QuickhullMesh") + std::to_string(procCount);
+
+        procCount++;
+
+        core::assets::asset<mesh> meshAsset = core::assets::AssetCache<mesh>::create(name,newMesh);
```

Comment:
yeah i'd recommend getting used to using lot's of auto, but ig this is prolly pretty old code since i do see you using auto everywhere else XD